src/data/CassandraBackend.hpp

```diff
@@ -567,6 +567,32 @@ class BasicCassandraBackend : public BackendInterface {
         return std::nullopt;
     }
 
+    std::vector<Blob>
+    doFetchLastTwoLedgerObjects(ripple::uint256 const& key, std::uint32_t const sequence, boost::asio::yield_context yield)
+        const override
+    {
+        LOG(log_.debug()) << "Fetching last two ledger objects for seq " << sequence << ", key = " << ripple::to_string(key);
+        if (auto const res = executor_.read(yield, schema_->selectLastTwoObjects, key, sequence); res) {
+            auto const& result = res.value();
+            if (not result.hasRows()) {
+                LOG(log_.error()) << "Could not fetch last two ledger objects - no rows";
+                return {};
+            }
+            std::vector<Blob> objects;
+            for (auto [object] : extract<Blob>(result))
+                objects.push_back(std::move(object));
+            if (objects.size() != 2) {
```

Comment:
It's completely possible that the size returned is less than or equal to 2. 
Ex. the object has not been deleted, or the object doesn't exist in the database

Reply:
It would be good to start thinking about the different possibilities when you start working on the next ticket.

Possible things:
1. Two entries are returned and last one is empty blob
2. two entries are returned but both are non-empty
3. one entry is returned and it is non-empty
4. one entry is return and it is empty (though i don't think this situation is possible, but still good to check in the code)
5. no entry is returned.